clinica/utils/filemanip.py

```diff
@@ -16,7 +15,7 @@ def _zip_unzip_nii(in_file: str, same_dir: bool, compress: bool):
     if (in_file is None) or isinstance(in_file, _Undefined):
         return None
 
-    if isinstance(in_file, Iterable):
+    if isinstance(in_file, (list, tuple)):
         return [_zip_unzip_nii(f, same_dir, compress) for f in in_file]
 
     in_file = Path(in_file)
```

Comment:
We might be able to simplify the whole None, str, and sequence-type handling with:

```python
    try:
        # Assuming in_file is path-like.
        in_file = Path(in_file)
    except TypeError:
        try:
            # Assuming in_file is a sequence type.
            return [_zip_unzip_nii(f, same_dir, compress) for f in in_file]
        except TypeError:
            # All other cases.
            return None
```

Reply:
Thanks! Much better indeed 👍